core/shared/src/main/scala/scalaz/zio/RefM.scala

```diff
@@ -70,12 +70,15 @@ object RefM extends Serializable {
       interrupted.get.flatMap {
         case Some(ts) => onDefect(ts)
         case None =>
-          update(a).sandboxed.redeem({
-            case Left(ts) => onDefect(ts)
-            case Right(n) => n
-          }, {
-            case (b, a) => ref.set(a) <* promise.complete(b)
-          })
+          update(a).sandboxed.redeem(
+            cause =>
+              cause.unchecked match {
+                case Nil     => IO.unit
+                case defects => onDefect(defects)
```

Comment:
Can `onDefect` become a `Cause[E] => IO[Nothing, Unit]`? Then this becomes `cause.fold(IO.unit, onDefect)` if I see this correctly.

Reply:
The other call to `onDefect` (a few lines up) is based on the result of `onTermination` which also take `List[Throwable] => IO[Nothing, Unit]`. I guess those two should be aligned?

Reply:
Oh yeah good spot, specifically `Cause[Nothing]`!

Reply:
In fact, `onTermination` is the only remaining occurrence of `List[Throwable]` in `IO` 🎉

Reply:
Haha, just checked the same thing ;)